### PR TITLE
Got paddle movement boundaries working.

### PR DESCRIPTION
**Pull request by Lil-P**
Also edited displacement calculation to reduce struct references.
Implemented balls speeding up on baddle collision.
Need to fix ball-paddle collisions not working at higher speeds (possibly use sleeping to control speed instead).

Signed-off-by: Philip Christoffersen <Philip@LAPTOP-Q9OEUBTF>

### CODE REVIEW

File: Workspace/Lab5/Game/Game.c [hunk 1]
```diff
@@ -331,24 +332,24 @@ void ReadJoystickHost()
 
         G8RTOS_WaitSemaphore(&PlayerMutex);
         {
-            /* Modify displacement based on joystick position */
-            updateDisplacement(&x_coord);
+            /* Modify displacement based on JoyStick position */
+            int16_t displacement = GameState.player.displacement;
```
Comment (wespiard) [inline Workspace/Lab5/Game/Game.c]: How about int16_t displacement = GameState.player.displacement + updateDisplacement(x_coord);

Comment (Lil-P) [reply Workspace/Lab5/Game/Game.c]: true true

File: Workspace/Lab5/Game/Game.c [hunk 2]
```diff
@@ -331,24 +332,24 @@ void ReadJoystickHost()
...
+            displacement += updateDisplacement(x_coord);
 
-            int16_t center = GameState.players[BOTTOM].currentCenter;
-            int16_t displacement  = GameState.player.displacement;
-            int16_t new_center = center + displacement;
+            int16_t center       = GameState.players[BOTTOM].currentCenter;
+            int16_t new_center   = center + displacement;
```
Comment (wespiard) [inline Workspace/Lab5/Game/Game.c]: If the center variable isn't used anywhere else we can do:
int16_t new_center = GameState.player[BOTTOM].currentCenter + displacement;

Comment (Lil-P) [reply Workspace/Lab5/Game/Game.c]: Yeah, I don't think it is

**Pull request merged**